duckduckgo/iOS
# Manage Favorites from Favorites Overlay

Task/Issue URL: https://app.asana.com/0/414709148257752/1203858036570302/f

### Description:
We allow editing and reordering favorites from the HomeViewController and the Bookmarks Manager but not from the Favorites Overlay.

This change allows Reordering, Editing, and Removing Favorites from the Favorites Overlay, and removes logic around `textFieldDidEndEditing(_:)` for the Omnibar to prevent the Favorites view from being dismissed while interacting with favorites.

#### Bonus:
Changes different strings in MainViewController to Constants

### Steps to test this PR:
1. Fire up the app (Make sure you have a few 'Favorites' added)
2. Navigate to any website (Important ⚠️)
3. Tap the Omnibar
4. Long-Press on one of your favorites
5. Observe that the 'Edit' / 'Delete' popover appears
6. **Test the functionality of both 'Edit' and 'Delete' from that view.**

7. Navigate to another website
8. Tap the omnibar
9. Reorder Your favorites via drag/drop
10. Tap 'Cancel.'
11. Tap the omnibar
12. **Observe that Favorites are ordered as expected**

13. Navigate to a few more websites
14. Tap the Ombibar
15. Tap the 'Back' button
16. **Observe the Omnibar is dismissed and the cancel button is no longer visible**
17. Repeat these steps 13-16 with all buttons in the Tab Bar.

### Demo
![demo](https://user-images.githubusercontent.com/1156669/226630082-973bafde-9d79-4cc3-8e4e-51f31860e1f3.gif)



---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
